### PR TITLE
Standalone iOS UI Tester app that enable admob UI Test Automation

### PR DESCRIPTION
**Pull request by sunmou99**
### Description
Standalone iOS UI Tester app that enable admob UI Test Automation.
All UI Test in `FirebaseCppUITestAppUITests.swift` are temporary Test code. I may change it later
***

[replace this line]: # (Describe your testing in detail.)
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [x] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

File: scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestApp/AppDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
+//
+//  AppDelegate.swift
+//  FirebaseCppUITestApp
+//
+//  Created by Mou Sun on 3/19/22.
+//  Copyright © 2022 Google. All rights reserved.
+//
+
```
Comment (DellaBitta) [inline scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestApp/AppDelegate.swift:8]: This should have our standard copyright notice.  See an [example swift file in the iOS Github Repo](https://github.com/firebase/firebase-ios-sdk/blob/8f5e7f0b376279f40ded653e04730fa2b23e0a89/FirebaseMessaging/Apps/SwiftUISample/SwiftUISample/ContentView.swift).

File: scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestApp/AppDelegate.swift [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  func application(
+    _ application: UIApplication,
+    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
+  ) -> Bool {
+    // Override point for customization after application launch.
+    return true
+  }
+
+  func applicationWillResignActive(_ application: UIApplication) {
+    // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
```
Comment (DellaBitta) [inline scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestApp/AppDelegate.swift:25]: Check the line length on these. comments.

File: scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestApp/ViewController.swift [hunk 3]
```diff
@@ -0,0 +1,18 @@
+//
+//  ViewController.swift
+//  FirebaseCppUITestApp
+//
+//  Created by Mou Sun on 3/19/22.
+//  Copyright © 2022 Google. All rights reserved.
+//
```
Comment (DellaBitta) [inline scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestApp/ViewController.swift:7]: copyright & license notice.

File: scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestAppUITests/FirebaseCppUITestAppUITests.swift [hunk 4]
```diff
@@ -0,0 +1,112 @@
+//
+//  FirebaseCppUITestAppUITests.swift
+//  FirebaseCppUITestAppUITests
+//
+//  Created by Mou Sun on 3/19/22.
+//  Copyright © 2022 Google. All rights reserved.
+//
```
Comment (DellaBitta) [inline scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestAppUITests/FirebaseCppUITestAppUITests.swift:7]: copyright & license notice.

File: scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestAppUITests/FirebaseCppUITestAppUITests.swift [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    let app = XCUIApplication(bundleIdentifier: "com.google.ios.admob.testapp")
+    app.launch()
+
+    // TestAdViewAdClick
+    var reference = app.staticTexts["Test mode"]
+    XCTAssertTrue(reference.waitForExistence(timeout: 60))
+    // click on the center point of the "Test Ad" TextView, where the Ad present
+    var x = (reference.frame.origin.x + reference.frame.width)/2
+    var y = (reference.frame.origin.y + reference.frame.height)/2
+    sleep(5)  // wait until button hittable
```
Comment (DellaBitta) [inline scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestAppUITests/FirebaseCppUITestAppUITests.swift:40]: nit: "wait until the button is clickable"

File: scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestAppUITests/FirebaseCppUITestAppUITests.swift [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    sleep(5)
+
+    // TestInterstitialAdClose
+    reference = app.staticTexts["Test mode"]
+    XCTAssertTrue(reference.waitForExistence(timeout: 60))
+    // click the top left corner close bottom.
+    // Use "Test Ad" TextView bottom position as the reference
+    x = (reference.frame.origin.y + reference.frame.height)/2
+    y = (reference.frame.origin.y + reference.frame.height)/2
+    sleep(5)  // wait until button hittable
```
Comment (DellaBitta) [inline scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestAppUITests/FirebaseCppUITestAppUITests.swift:55]: nit: "wait until the button is clickable"

File: scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestAppUITests/FirebaseCppUITestAppUITests.swift [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+    sleep(5)
+
+    // TestInterstitialAdClickAndClose
+    reference = app.staticTexts["Test mode"]
+    XCTAssertTrue(reference.waitForExistence(timeout: 60))
+    // click the center point of the device, where the Ad present
+    x = app.frame.width/2
+    y = app.frame.height/2
+    sleep(5)  // wait until button hittable
```
Comment (DellaBitta) [inline scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestAppUITests/FirebaseCppUITestAppUITests.swift:67]: nit: "wait until the button is clickable"

File: scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestAppUITests/FirebaseCppUITestAppUITests.swift [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+    sleep(5)
+
+    // TestRewardedAdClose
+    reference = app.webViews.staticTexts.containing(NSPredicate(format: "label CONTAINS 'seconds'")).element
+    XCTAssertTrue(reference.waitForExistence(timeout: 60))
+    // click the top right corner close bottom.
+    // Use "* seconds" TextView right position as the reference
+    x = (reference.frame.origin.x + reference.frame.width + app.frame.width)/2
+    y = (reference.frame.origin.y + reference.frame.height)/2
+    sleep(5)  // wait until button hittable
```
Comment (DellaBitta) [inline scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestAppUITests/FirebaseCppUITestAppUITests.swift:84]: nit: "wait until the button is clickable"

**Review by DellaBitta — CHANGES_REQUESTED** [review]
Copyright & license comments, and a few nits.

File: scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestAppUITests/FirebaseCppUITestAppUITests.swift (hunk 5 above)
Comment (sunmou99) [reply scripts/gha/ui_testing/uitest_apple/FirebaseCppUITestAppUITests/FirebaseCppUITestAppUITests.swift:40]: Apple Developer Document uses hittable
https://developer.apple.com/documentation/xctest/xcuielement/1500561-hittable?language=objc

**Review by DellaBitta — APPROVED** [review]

**Pull request merged**